### PR TITLE
Fix view rotation when rendering in background mode

### LINKED ISSUE #1446: can't switch to portrait mode while in background mode
```python
import numpy
data=numpy.sin(numpy.arange(100))
data=numpy.reshape(data,(10,10))
import vcs
x=vcs.init()
x.plot(data,bg=1)
x.clear()
x.portrait()
```
leads to:
```
python: /home/doutriaux1/build/build/VTK/Rendering/OpenGL/vtkOpenGLRenderWindow.cxx:1603: int vtkOpenGLRenderWindow::CreateHardwareOffScreenWindow(int, int): Assertion `"pre: positive_width" && width>0' failed.
```

### PR DESCRIPTION
**Pull request by sankhesh**
Fixes #1446

### CODE REVIEW

**Comment by sankhesh** [comment]
@doutriaux1 @aashish24

**Comment by doutriaux1** [comment]
@sankhesh fails on mac?

**Comment by sankhesh** [comment]
@doutriaux1 

I think I know the issue. I'll push a fix soon.

**Comment by aashish24** [comment]
Looks like mac is running into build issues :cry:

**Comment by aashish24** [comment]
CDMS_Test_multiple_formats because of opendap issue.

**Comment by doutriaux1** [comment]
@sankhesh fails on mac and RH6 w/o MESA

**Comment by doutriaux1** [comment]
@aashish24 CDMS_Test_multiple_formats is weird it actually set faults on this machine. But it's not related to the regular .dodsrc open dap issue.

**Comment by durack1** [comment]
@doutriaux1 it's [vcs_test_background_mode_rotate](https://open.cdash.org/testSummary.php?project=44&name=vcs_test_background_mode_rotate&date=2015-09-23) this time

**Comment by durack1** [comment]
@doutriaux1 looks like [`Xvfb` needs a restart on `crunchy`](https://open.cdash.org/testDetails.php?test=376395732&build=4026980)
```
Successfully updated your environment to use UVCDAT
(changes are valid for this session/terminal only)
Version: 2.2.0-538-g1922557
Location: /export/doutriaux1/uvcbot/build/install
failed to create drawable
X Error of failed request:  GLXBadContextTag
  Major opcode of failed request:  149 (GLX)
  Minor opcode of failed request:  5 (X_GLXMakeCurrent)
  Serial number of failed request:  15662
  Current serial number in output stream:  15662)
```

**Comment by durack1** [comment]
@doutriaux1 and the same weird issue with graphic output with fills outside of the box border - the output of the test on `garant`:
displayimage php

**Comment by doutriaux1** [comment]
@durack1 this is a new test and it fails on the mac as wel.. I can restart Xvfb but I doubt that's the issue

File: Packages/vcs/Lib/VTKPlots.py [hunk 1]
```diff
@@ -405,6 +408,12 @@ def portrait(self, W, H, x, y, clear):
         else:
             self.renWin.SetSize(W, H)
 
+    def portrait(self, W=-99, H=-99, x=-99, y=-99, clear=0):
+        self.resize_or_rotate_window(W, H, x, y, clear)
+
```
Comment (doutriaux1) [inline Packages/vcs/Lib/VTKPlots.py]: @sankhesh this does not work the following snippet demonstrates the problem:
```python
import cdms2
import vcs
import os
f=cdms2.open(os.path.join(vcs.sample_data,"clt.nc"))
s=f("clt")
x=vcs.init()
x.portrait()  # turn page in portrait mode
x.portrait()  # turn page in portrait mode (again but it is already in portrait so it should be doing nothing)
x.plot(s)
x.png("test_portrait_bg")
```

leads to:
test_portrait_bg

Essentially you rotate twice

Comment (doutriaux1) [inline Packages/vcs/Lib/VTKPlots.py]: @sankhesh you might want to add this as a test too if @aashish24 thinks it is a worthy test :wink:

Comment (doutriaux1) [inline Packages/vcs/Lib/VTKPlots.py]: @sankhesh I pushed a fix

Comment (doutriaux1) [inline Packages/vcs/Lib/VTKPlots.py]: @sankhesh sorry there is still an issue somewhere:
```python
import vcs
import cdms2
import os

f=cdms2.open(os.path.join(vcs.sample_data,"clt.nc"))
s=f("clt",slice(0,1))
x=vcs.init()
x.plot(s,bg=1)
x.png("landscape")
x.clear()
x.portrait()
x.plot(s,bg=1)
x.png("portrait")
```
comes out landscape

**Comment by durack1** [comment]
@doutriaux1 looks like the [`vcs_addons_*` baselines might be broken](https://open.cdash.org/viewTest.php?onlyfailed&buildid=4027790) here?

**Comment by sankhesh** [comment]
@doutriaux1 @durack1 @aashish24 My latest changes fixed the background_rotate test that was added new in this branch.

**Comment by doutriaux1** [comment]
@sankhesh does this fix the second script i posted last night

**Comment by sankhesh** [comment]
@doutriaux1  Yes

**Comment by doutriaux1** [comment]
@sankhesh your eztemplate png are actually better, can you please add them to the new baselines?

**Comment by sankhesh** [comment]
@doutriaux1 What do you mean by ***better***? All the EzTemplate tests pass on my machine i.e. they generate the same images as the baselines. What am I missing something here?

**Comment by aashish24** [comment]
@jbeezley when I visit a link to garant in here, it does not show the page I am expected to see. Is that a bug?

**Comment by jbeezley** [comment]
If the build stage doesn't finish, then the cdash link won't work.  You need to look for the build manually.  C.f. https://open.cdash.org/viewBuildError.php?buildid=4029068  It is one of the things that would be nice to have an API for in CDash.

**Comment by jbeezley** [comment]
Actually, I was thinking this was the libx264 PR.  It looks like the difference between a "working" link and a "broken" link the `=N` at the end.  I have no idea why the buildbot puts that there or what it is supposed to me.

https://open.cdash.org/index.php?compare1=65&filtercount=2&field1=buildname%2Fstring&project=UV-CDAT&field2=buildstarttime%2Fdate&showfilters=0&limit=100&compare2=83&value1=f2ee8f46-build1214-uvcdat-garant-linux-release&showfeed=0&display=project&value2=N

https://open.cdash.org/index.php?compare1=65&filtercount=2&field1=buildname%2Fstring&project=UV-CDAT&field2=buildstarttime%2Fdate&showfilters=0&limit=100&compare2=83&value1=f2ee8f46-build1214-uvcdat-garant-linux-release&showfeed=0&display=project&value2

**Comment by doutriaux1** [comment]
@sankhesh on my Mac running your branch. The pngs are in portrait mode whereas it seems that the baselines png are in landscape. No?

**Comment by doutriaux1** [comment]
@sankhesh see https://open.cdash.org/viewTest.php?onlyfailed&buildid=4035541

**Comment by doutriaux1** [comment]
@sankhesh could it be the bit that gets the screen size to determine the window size?

**Comment by aashish24** [comment]
@doutriaux1 it does looks like that they are in different mode.

**Comment by doutriaux1** [comment]
@sankhesh should we worry about crunchy failure? Maybe post rc1?

**Comment by sankhesh** [comment]
@doutriaux1 Not sure why there is a GLX issue on crunchy just for this one test. Any ideas? 

I am assuming you are referring to my [EzTemplate] (https://github.com/UV-CDAT/uvcdat/pull/1557#issuecomment-143219423) question about the portrait vs landscape images. Yes, it does seem like your mac is generating portrait images. I think it is related to your latest screensize changes.

**Comment by doutriaux1** [comment]
@sankhesh all platforms pass. Merging in.

**Pull request merged**